src/settings/storeconf.h

```diff
@@ -14,37 +14,39 @@
 #ifndef STORECONF_H
 #define STORECONF_H
 
-#include <QWidget>
 #include <QJsonObject>
 #include <QJsonDocument>
+#include <QStringList>
+#include <QMap>
+#include <QFile>
 
-class StoreConf : public QWidget
+class StoreConf
 {
-    Q_OBJECT
 public:
-    explicit StoreConf(QWidget * parent = nullptr);
+    explicit StoreConf(QString userName = "unnamed");
+
+    void restoreConFile();          //  restoring settings from last session
+    void saveConFile();             //  saving settings at the end of current session
 private:
-    QString getPathToConFile(); // search configuration file. if not exist - create it with default parameters
-    QString conFile;
-    QString conFileDefault          = "conf.json";
-
-    QString organizationName        = "Lv-425.C++";
-    QString applicationVersion      = "0.1";
-    QString applicationName         = "Pair Storm";
-    QString analizerStyle           = "white";
-    QString analizerFontSize        = "12";
-    QString analizerFontName        = "Courier 10 Pitch";
-    QString cppExtentions           = ".c;.cpp;.h;.hpp;.json;.txt";
-    QStringList cppExtentionsList {".c", ".cpp", ".h", ".hpp", ".json", ".txt"};
-
-    void writeJson();               //  if conf.json not exist - creates it in current directory
-    void readJson();                //  read [confile] to [loadDoc]. status write to [readStatus]
+    void getPathToConFile();        //  search configuration file. if not exist - create it with default parameters
+
+    QString mConFile;               //  name of configuration file
+    QString mPathToConFile;         //  path to configuration file
+    QMap<QString, QString> mFields; //  first -> valueName, second -> valueItself
+
+    QStringList mCppExtentionsList; //  all extentions available in C++ mode
+
+    void checkConfDir(QString str = "exe");    //  check existance directory <conf>, create if not exist
+                                               //  str = "exe" - directory in same dir as execution file
+    void writeJson(QString str = "start");     //  on application start: if conf.json not exist - creates it in conf directory
+                                               //  on application quit:  rewrite json with values from QSettings
+    void readJson();                //  read configuration file to [loadDoc]. status write to [readStatus]
     void parseJson();
```

Comment:
readJson & writeJson both indicate execution status using flags mReadStatus/mWriteStatus
To my opinion it is better to use exceptions (C++ approach) or at least return bool (C approach)

Also I don't see if you use mWriteStatus at all

Reply:
ok
I remove mWriteStatus